### PR TITLE
 handle click item in genres

### PR DESCRIPTION
**Pull request by meoconlontonbebong**
Link redmi : https://edu-redmine.framgia.vn/issues/6811
 Handle click item in genres .

### CODE REVIEW

**Comment by meoconlontonbebong** [comment]
ready

File: app/src/main/java/com/example/moviedb_37/screen/home/HomeFragment.java [hunk 1]
```diff
@@ -91,24 +93,31 @@ public void onDestroy() {
...
-        bundle.putString(BUNDLE_KEY, String.valueOf(genre.getId()));
-        bundle.putString(BUNDLE_NAME, genre.getName());
-        showMovies(bundle);
+    public void onGenreItemClick(Genre genre) {
+        showMovies(genre, GENRE_SOURCE);
+    }
+
+    @Override
+    public void showMovies(Genre genre, int getBy) {
+        Intent intent = ListMovieCategoryActivity.getMoviesIntent(getActivity(), genre, getBy);
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/moviedb_37/screen/home/HomeFragment.java]:
```suggestion
        Intent intent = ListMovieCategoryActivity.getIntent(getActivity(), genre, getBy);
```

File: app/src/main/java/com/example/moviedb_37/screen/home/HomeFragment.java [hunk 2]
```diff
@@ -91,24 +93,31 @@ public void onDestroy() {
...
+        startActivity(intent);
+    }
+
+    @Override
+    public void showMovieDetail(Movie movie) {
+        startActivity(MovieDetailsActivity.getMovieDetailsIntent(getActivity(), movie));
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/moviedb_37/screen/home/HomeFragment.java]:
```suggestion
        startActivity(MovieDetailsActivity.getIntent(getActivity(), movie));
```

File: app/src/main/java/com/example/moviedb_37/screen/movies/ListMovieCategoryActivity.java [hunk 3]
```diff
@@ -118,4 +135,14 @@ protected void onDestroy() {
...
     }
+
+    @Override
+    public void onMovieItemClick(Movie movie) {
+        showMovieDetail(movie);
+    }
+
+    @Override
+    public void showMovieDetail(Movie movie) {
+        startActivity(MovieDetailsActivity.getMovieDetailsIntent(this, movie));
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/example/moviedb_37/screen/movies/ListMovieCategoryActivity.java]:
```suggestion
        startActivity(MovieDetailsActivity.getIntent(this, movie));
```

**Comment by meoconlontonbebong** [comment]
ready

**Pull request merged**